File: Examples/test-suite/perl5/template_ns_runme.pl
use strict;
use warnings;
use Test::More tests => 6;
BEGIN { use_ok('template_ns') }
require_ok('template_ns');

my $p1 = template_ns::pairii->new(2, 3);
my $p2 = template_ns::pairii->new($p1);

is($p2->{first}, 2);
is($p2->{second}, 3);

my $p3 = template_ns::pairdd->new(3.5, 2.5);
my $p4 = template_ns::pairdd->new($p3);

is($p4->{first}, 3.5);
is($p4->{second}, 2.5);

File: Examples/test-suite/lua/template_ns_runme.lua
local v=require("template_ns")

local p1 = v.pairii(2, 3)
local p2 = v.pairii(p1)
assert(p2.first == 2)
assert(p2.second == 3)

local p3 = v.pairdd(3.5, 2.5)
local p4 = v.pairdd(p3)
assert(p4.first == 3.5)
assert(p4.second == 2.5)
